### ISSUE TITLE
Incorrect results reported when pinging multiple hosts

### ISSUE THREAD

**Issue by karlism** [schweikert/fping#97]
Hello,

Incorrect number of transmitted and received packets is sometimes reported, when running fping with "-l" and "-Q" options, especially when latency to some of the hosts is big.

Here's an example fping output while pinging multiple hosts:
$ /tmp/fping-3.13/src/fping -v
fping: Version 3.13
fping: comments to david@schweikert.ch
$ /tmp/fping-3.13/src/fping -Q 10 -l -D -f /tmp/pinghosts
[11:15:58]
hkg-host01 : **xmt/rcv/%loss = 9/8/11%**, min/avg/max = 307/307/307
hkg-host02 : **xmt/rcv/%loss = 9/8/11%**, min/avg/max = 307/307/307
127.0.0.1      : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.04/0.06
127.0.0.2      : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.04/0.04/0.06
127.0.0.3      : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.04/0.04/0.06
127.0.0.4      : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.04/0.04/0.06
127.0.0.5      : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.05/0.06
127.0.0.6      : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.04/0.06
127.0.0.7      : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.04/0.06
127.0.0.8      : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.04/0.04/0.06
127.0.0.9      : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.04/0.04/0.05
127.0.0.10     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.04/0.05
127.0.0.11     : xmt/rcv/%loss = 8/8/0%, min/avg/max = 0.03/0.04/0.05
127.0.0.12     : xmt/rcv/%loss = 8/8/0%, min/avg/max = 0.03/0.04/0.06
127.0.0.13     : xmt/rcv/%loss = 8/8/0%, min/avg/max = 0.03/0.04/0.06
127.0.0.14     : xmt/rcv/%loss = 8/8/0%, min/avg/max = 0.03/0.04/0.06
127.0.0.15     : xmt/rcv/%loss = 8/8/0%, min/avg/max = 0.03/0.04/0.05
127.0.0.16     : xmt/rcv/%loss = 8/8/0%, min/avg/max = 0.04/0.05/0.07
127.0.0.17     : xmt/rcv/%loss = 8/8/0%, min/avg/max = 0.03/0.04/0.06
127.0.0.18     : xmt/rcv/%loss = 8/8/0%, min/avg/max = 0.04/0.05/0.06
127.0.0.19     : xmt/rcv/%loss = 8/8/0%, min/avg/max = 0.03/0.04/0.05
127.0.0.20     : xmt/rcv/%loss = 8/8/0%, min/avg/max = 0.04/0.04/0.05
mlt-host01  : xmt/rcv/%loss = 8/8/0%, min/avg/max = 83.1/83.2/83.3
mlt-host02  : xmt/rcv/%loss = 8/8/0%, min/avg/max = 83.2/83.3/84.2
127.0.0.21     : xmt/rcv/%loss = 8/8/0%, min/avg/max = 0.03/0.04/0.05
127.0.0.22     : xmt/rcv/%loss = 8/8/0%, min/avg/max = 0.04/0.05/0.06
127.0.0.23     : xmt/rcv/%loss = 8/8/0%, min/avg/max = 0.03/0.04/0.06
127.0.0.24     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.04/0.06
127.0.0.25     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.04/0.06
127.0.0.26     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.04/0.06
127.0.0.27     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.04/0.04/0.06
127.0.0.28     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.04/0.04/0.06
127.0.0.29     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.04/0.06
127.0.0.30     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.04/0.06
127.0.0.31     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.04/0.06
127.0.0.32     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.04/0.04/0.06
127.0.0.33     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.04/0.04/0.05
127.0.0.34     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.04/0.09
127.0.0.35     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.05/0.06
127.0.0.36     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.04/0.06
127.0.0.37     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.04/0.04/0.05
127.0.0.38     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.04/0.05
127.0.0.39     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.04/0.05/0.08
127.0.0.40     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.04/0.04/0.05
127.0.0.41     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.04/0.05/0.08
127.0.0.42     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.04/0.06/0.12
[11:16:08]
hkg-host01    : **xmt/rcv/%return = 8/9/112%**, min/avg/max = 307/307/308
hkg-host02    : **xmt/rcv/%return = 8/9/112%**, min/avg/max = 307/307/307
127.0.0.1      : xmt/rcv/%loss = 8/8/0%, min/avg/max = 0.03/0.04/0.05
127.0.0.2      : xmt/rcv/%loss = 8/8/0%, min/avg/max = 0.03/0.04/0.09
127.0.0.3      : xmt/rcv/%loss = 8/8/0%, min/avg/max = 0.03/0.04/0.06
127.0.0.4      : xmt/rcv/%loss = 8/8/0%, min/avg/max = 0.03/0.04/0.06
127.0.0.5      : xmt/rcv/%loss = 8/8/0%, min/avg/max = 0.03/0.05/0.13
127.0.0.6      : xmt/rcv/%loss = 8/8/0%, min/avg/max = 0.03/0.03/0.05
127.0.0.7      : xmt/rcv/%loss = 8/8/0%, min/avg/max = 0.03/0.03/0.05
127.0.0.8      : xmt/rcv/%loss = 8/8/0%, min/avg/max = 0.03/0.04/0.06
127.0.0.9      : xmt/rcv/%loss = 8/8/0%, min/avg/max = 0.04/0.04/0.06
127.0.0.10     : xmt/rcv/%loss = 8/8/0%, min/avg/max = 0.03/0.03/0.05
127.0.0.11     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.04/0.06
127.0.0.12     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.05/0.13
127.0.0.13     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.02/0.04/0.06
127.0.0.14     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.03/0.05
127.0.0.15     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.04/0.05
127.0.0.16     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.04/0.06
127.0.0.17     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.04/0.05
127.0.0.18     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.02/0.03/0.05
127.0.0.19     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.04/0.06
127.0.0.20     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.04/0.05
mlt-host01     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 83.2/83.8/87.3
mlt-host02     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 83.2/83.3/83.4
127.0.0.21     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.04/0.04/0.06
127.0.0.22     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.04/0.07
127.0.0.23     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.05/0.11
127.0.0.24     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.04/0.04/0.06
127.0.0.25     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.04/0.04/0.05
127.0.0.26     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.02/0.04/0.09
127.0.0.27     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.04/0.05
127.0.0.28     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.04/0.05
127.0.0.29     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.04/0.04/0.05
127.0.0.30     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.04/0.12
127.0.0.31     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.05/0.15
127.0.0.32     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.04/0.04/0.07
127.0.0.33     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.04/0.06
127.0.0.34     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.04/0.06
127.0.0.35     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.04/0.05
127.0.0.36     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.04/0.05
127.0.0.37     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.04/0.04/0.05
127.0.0.38     : xmt/rcv/%loss = 9/9/0%, min/avg/max = 0.03/0.05/0.08
127.0.0.39     : xmt/rcv/%loss = 8/8/0%, min/avg/max = 0.04/0.04/0.06
127.0.0.40     : xmt/rcv/%loss = 8/8/0%, min/avg/max = 0.03/0.04/0.05
127.0.0.41     : xmt/rcv/%loss = 8/8/0%, min/avg/max = 0.04/0.04/0.06
127.0.0.42     : xmt/rcv/%loss = 8/8/0%, min/avg/max = 0.03/0.03/0.04

If 127.0.0.0/8 test hosts are removed from the list, then fping behaves as expected.

**Comment by karlism** [comment]
There is also an issue with max RTT response reporting while pinging multiple hosts simultaneously:
fping -Q 10 -l -T 1 -p 5 -i 5 -s -D -f pinghosts
[07:51:51]
10.10.10.10 : xmt/rcv/%loss = 268/268/0%, min/avg/max = 0.29/1.07/**20.1**
10.10.10.11 : xmt/rcv/%loss = 268/268/0%, min/avg/max = 0.31/1.17/**17.2**
10.10.10.14 : xmt/rcv/%loss = 267/267/0%, min/avg/max = 0.30/0.84/**11.0**
10.10.10.12 : xmt/rcv/%loss = 267/267/0%, min/avg/max = 0.31/0.91/**11.0**
10.10.10.13 : xmt/rcv/%loss = 267/267/0%, min/avg/max = 0.29/0.99/**10.9**
10.10.10.80 : xmt/rcv/%loss = 267/267/0%, min/avg/max = 0.39/1.07/**11.0**
10.10.10.81 : xmt/rcv/%loss = 267/267/0%, min/avg/max = 0.41/1.20/**11.0**
10.10.10.10 : xmt/rcv/%loss = 299/299/0%, min/avg/max = 0.29/1.03/**20.1**
10.10.10.11 : xmt/rcv/%loss = 299/299/0%, min/avg/max = 0.31/1.12/**17.2**
10.10.10.14 : xmt/rcv/%loss = 298/298/0%, min/avg/max = 0.30/0.80/**11.0**
10.10.10.12 : xmt/rcv/%loss = 298/298/0%, min/avg/max = 0.31/0.87/**11.0**
10.10.10.13 : xmt/rcv/%loss = 298/298/0%, min/avg/max = 0.29/0.95/**10.9**
10.10.10.80 : xmt/rcv/%loss = 298/298/0%, min/avg/max = 0.39/1.03/**11.0**
10.10.10.81 : xmt/rcv/%loss = 298/298/0%, min/avg/max = 0.41/1.18/**11.0**

       7 targets
       7 alive
       0 unreachable
       0 unknown addresses

       0 timeouts (waiting for response)
    2088 ICMP Echos sent
    2088 ICMP Echo Replies received
      11 other ICMP received

 0.29 ms (min round trip time)
 1.00 ms (avg round trip time)
 **20.1** ms (max round trip time)
       11.110 sec (elapsed real time)

But pinging hosts with _ping_ at the same time reports that max RTT never exceeds 5ms, even when pinging with larger packet size:

$ ping -i .1 10.10.10.10 -s 1000
PING 10.10.10.10 (10.10.10.10) 1000(1028) bytes of data.
1008 bytes from 10.10.10.10: icmp_seq=1 ttl=126 time=0.515 ms
[...]
1008 bytes from 10.10.10.10: icmp_seq=80 ttl=126 time=0.494 ms
--- 10.10.10.10 ping statistics ---
80 packets transmitted, 80 received, 0% packet loss, time 7975ms
rtt min/avg/max/mdev = 0.476/1.161/4.523/0.944 ms

**Comment by schweikert** [comment]
This a bit of a conceptual problem because of the fact that there are various loops/timers involved. See also the discussion in #105.
The proper solution would be making sure that probes sent shortly before the report interval are only used for the next report, but this probably requires substantial refactoring.

**Comment by schweikert** [comment]
Anyway, could you test with the current develop branch and with parameters like -p 1007 -Q 10
